lib/css/components/notices.less

```diff
@@ -1,252 +1,267 @@
-.construct-notice-component() {
-    background: var(--_no-bg, var(--black-050));
-    border-color: var(--_no-bc, var(--bc-medium));
-    border-style: solid;
-    color: var(--_no-fc, var(--fc-medium));
-    font-size: var(--fs-body1);
-
-    code {
-        background: var(--_no-code-bg, transparent);
-    }
-
-    & &--btn {
-        color: inherit;
-        padding: var(--su8);
-
-        &:focus,
-        &:hover {
-            background: var(--_no-btn-focus-bg, var(--black-100));
-        }
-
-        &:active {
-            background: var(--_no-btn-active-bg, var(--black-150));
-        }
-    }
+.construct-notice-component(@baseClass) {
+    // BASE COMPONENT-SPECIFIC CUSTOM PROPERTIES
+    --_no-bc: var(--bc-medium);
+    --_no-bg: var(--black-050);
+    --_no-fc: var(--fc-medium);
+    --_no-btn-bg-focus: var(--black-100);
```

Comment:
I haven't decided on a naming convention for these custom properties that affect both `:focus` and `:hover`. I'm also not sure it matters.